[JSQMessagesViewController] Upgrade to version v7.3.4

vlesierse opened the pull request:
This PR includes the upgrade of JSQMessagesViewController from version 7.2.0 to 7.3.4.
[JSQMessagesViewController 7.3.4](https://github.com/jessesquires/JSQMessagesViewController/releases/tag/7.3.4)

- Update ApiDefinition & Additions
- Change JSQMessagesViewControllerLib project
- Added AudioMessage to XamarinChat example

dnfclas commented:
@vlesierse,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

bholmes commented on iOS/JSQMessagesViewController/externals/Makefile:
```diff
@@ -8,20 +8,10 @@ JSQ_TARGET=JSQMessagesViewControllerLib
-	cd ios/JSQSystemSoundPlayer && git checkout 2.0.1
-
-ios/JSQMessagesViewController/JSQMessagesViewController/JSQMessages.h : ios/JSQSystemSoundPlayer/JSQSystemSoundPlayer/Classes/JSQSystemSoundPlayer.h 
+ios/JSQMessagesViewController/JSQMessagesViewController/JSQMessages.h :
```
JSQSystemSoundPlayer can not be removed.  It will cause linker errors for release device builds.  Test they by building the samples in Release for iPhone.  Can you put JSQSystemSoundPlayer back in the Makefile?

vlesierse replied:
Sure, will move it back. Although it seems not the be a problem in my environment. Will try it with a clean repo.

vlesierse replied:
Using a clean repo allowed me to build and test the samples apps in both Debug and Release mode. Can you check it again?

vlesierse replied:
@bholmes Ok, I found a reproducible scenario and reverted the change. It should be all good now.

bholmes commented on iOS/JSQMessagesViewController/externals/ios/JSQMessagesViewControllerLib/JSQMessagesViewControllerLib.xcodeproj/project.pbxproj:
```diff
@@ -21,7 +20,6 @@
 		2188DD431AD876C30043EFEE /* JSQMessagesMediaViewBubbleImageMasker.m in Sources */ = {isa = PBXBuildFile; fileRef = 2188DCF51AD876C30043EFEE /* JSQMessagesMediaViewBubbleImageMasker.m */; };
 		2188DD441AD876C30043EFEE /* JSQMessagesTimestampFormatter.m in Sources */ = {isa = PBXBuildFile; fileRef = 2188DCF71AD876C30043EFEE /* JSQMessagesTimestampFormatter.m */; };
 		2188DD451AD876C30043EFEE /* JSQMessagesToolbarButtonFactory.m in Sources */ = {isa = PBXBuildFile; fileRef = 2188DCF91AD876C30043EFEE /* JSQMessagesToolbarButtonFactory.m */; };
-		2188DD461AD876C30043EFEE /* JSQSystemSoundPlayer.m in Sources */ = {isa = PBXBuildFile; fileRef = 2188DCFD1AD876C30043EFEE /* JSQSystemSoundPlayer.m */; };
```
Again the JSQSystemSoundPlayer files can not be removed.

bholmes commented on iOS/JSQMessagesViewController/samples/XamarinChat/MessageFactory.cs:
```diff
@@ -27,10 +27,12 @@ public class MessageFactory
 
-			if (choice == 0)
-				return CreateMonkeyMessage (user);
+            if (choice == 0)
```
Use tabs here not spaces

bholmes commented on iOS/JSQMessagesViewController/samples/XamarinChat/MessageFactory.cs:
```diff
@@ -60,6 +62,15 @@ Message CreateMonkeyMessage (User user)
+			NSUrl localFileUrl = NSBundle.MainBundle.GetUrlForResource("jsq_messages_sample", "m4a");
+			var audioData = NSData.FromUrl(localFileUrl);
+			var audioItem = new AudioMediaItem(audioData);
+            audioItem.AudioViewAttributes.PauseButtonImage = UIImageExtensions.DefaultPauseImage;
```
Tabs not spaces here too

bholmes reviewed (changes requested):
I have added comments.  Also please change the version number in the component/component.yaml file.

vlesierse committed:
Update to JSQMessagesViewController v7.3.4


Using tabs instead of spaces

Bumped version in component.yaml

Revert removal of JSQSystemSoundPlayer


Fix formatting problems with pbxproj file


Fix formatting problems Makefile


Another fix to Makefile

bholmes commented:
built and now in our release pipeline.  

https://jenkins.mono-project.com/view/Components/job/Components-OpenSource/45/
